gui/src/ruler.cc

```diff
@@ -0,0 +1,362 @@
+#include <cmath>
+#include <iostream>
+#include "ruler.hh"
+
+////////////////////////////////////////////////////////////////////////
+// Ruler
+
```

Comment:
Wow... This is file hard to read/understand. Don't have any concrete tips on how to fix that, though.